### ISSUE TITLE
All values are at 0 epoch

### ISSUE THREAD

**Issue by ShownX** [dmlc/tensorboard#46]
Hello,

I just use ```mx.contrib.tensorboard.LogMetricsCallback()``` in the ```model.fit(batch_end_callback=..., eval_batch_end_call_back=...)```

It generates the log file, but when I run ```tensorboard --logdir=...```,

it generates a figure like follows:

![screen shot 2017-06-15 at 11 33 59 am](https://user-images.githubusercontent.com/9678100/27189424-4c6ae67c-51bf-11e7-8592-c6021d80b957.png)

Any ideas what I did wrong?

**Comment by zihaolucky** [comment]
Try `relative` mode on the left.

**Comment by zihaolucky** [comment]
@ShownX As I haven't add `step` params in `scalar`, so you have to select the `relative` mode of TensorBoard.

<img width="333" alt="2017-06-16 3 18 42" src="https://user-images.githubusercontent.com/4514568/27216171-2059ce16-52a7-11e7-85fd-0b973e2a053f.png">

**Comment by ShownX** [comment]
Thank you very much!

**Comment by arundasan91** [comment]
@zihaolucky , Could you please guide me in the right direction on how to add the `step` param in `scalar` ? `Relative` mode is very bad as of now. Gives me very odd results in the training graph. 
![screen shot 2018-03-04 at 7 20 45 pm](https://user-images.githubusercontent.com/14352842/36953547-4d61f368-1fe1-11e8-9e9f-49a9123a8b2b.jpg)

I am calling tensorboard op on every batch_end.
```
batch_end_callbacks    += [mx.contrib.tensorboard.LogMetricsCallback(training_log)]
```
Tensorboard logs every batch but individually (no stitching between batches) giving `batches` number of graphs in the `Relative` mode.

**Comment by zihaolucky** [comment]
Hi @arundasan91 

The reason it looks ugly is we log train & valid/test data points with different time scale. You can write another callback function and pass the `step` explicitly.

**Comment by arundasan91** [comment]
Hi @zihaolucky, I was able to figure it out but forgot to update you. I passed the `params.epoch` to `global_steps` in tensorboard.py and it worked as intended. Thank you so much for the wonderful project!
Do you have any idea on why the `batch_end_caallback` gives discontinuous graphs? Some accuracy values are `nan` when I download the csv but prints out perfectly to shell while training.
